### PR TITLE
Caches data for paintings we have saved

### PR DESCRIPTION
**Pull request by dragomagol**
## About The Pull Request
We got some reports about a server lagging, the cause was determined to be TGUI. Potato figured out it was the painting manager, and I was able to recreate the lag by just quickly scrolling through the 1200+ paintings on Manuel via a curator console. Please don't do this btw.

## Why It's Good For The Game
Allow people to scrub through the painting gallery without lagging the server.

## Changelog
:cl: Tattle
fix: you should now be able to scrub through the library without lagging the server
/:cl:

### CODE REVIEW

File: code/controllers/subsystem/persistent_paintings.dm [hunk 1]
```diff
@@ -128,7 +133,7 @@ SUBSYSTEM_DEF(persistent_paintings)
 		"diamond" = PATRONAGE_AMAZING_FRAME,
 		"rainbow" = PATRONAGE_SUPERB_FRAME,
 		"supermatter" = PATRONAGE_LEGENDARY_FRAME
```
Comment (san7890) [inline code/controllers/subsystem/persistent_paintings.dm:135]:
```suggestion
		"supermatter" = PATRONAGE_LEGENDARY_FRAME,
```

if you're fixing indentations may as well make the full list perfect with the trailing comma

**Review by san7890 — COMMENTED** [review]
optional nit

File: code/controllers/subsystem/persistent_paintings.dm [hunk 2]
```diff
@@ -113,6 +113,11 @@ SUBSYSTEM_DEF(persistent_paintings)
 	/// A list of /datum/paintings saved or ready to be saved this round.
 	var/list/paintings = list()
 
+	/// A list of paintings' data for paintings that are currently stored in the library.
+	var/list/painting_data = list()
+	/// A list of paintings' data for paintings that are currently stored in the library, with admin metadata
+	var/list/admin_painting_data = list()
```
Comment (LemonInTheDark) [inline code/controllers/subsystem/persistent_paintings.dm:119]: what happens when a new painting is added?

Comment (dragomagol) [reply code/controllers/subsystem/persistent_paintings.dm]: Addressed in new commit (also made me realize that `painting_data` is used as a local variable elsewhere in the file)

**Comment by LemonInTheDark** [comment]
Potential for accidental massive lag, speed(ish) merging

**Pull request merged**